pubsub/Readme.md

```diff
@@ -25,3 +25,44 @@ type PubSub interface {
 	Subscribe(req SubscribeRequest, handler func(msg *NewMessage) error) error
 }
 ```
+
+### TTL (or Time To Live)
+
+Time to live is implemented by default in Dapr. A publishing application can add the `expiration` attribute to a CloudEvent object and Dapr runtime will drop the message if the expiration is due at the subscriber's end, prior to invoking subscriber applications. The `expiration` attribute is Dapr specific and extends the CloudEvent spec. This is a convenience to subscriber applications, so Dapr does not expire messages from pub subs that do not have subscribers via Dapr.
+
+Alternatively, the publisher application can set the expiration by publishing with the `ttlInSeconds` metadata. Components that support TTL should parse this metadata attribute. Dapr runtime will also automatically populate the `expiration` attribute in the CloudEvent object if `ttlInSeconds` is present - in this case, Dapr will add 1 second to allow component implementations to expire the message before Dapr consumes it.
```

Comment:
This 1 second delay is an arbitrary value to give time between the calculation of `expiration` in Dapr and the moment the event is published to the underlying component. If there is a big delay between those, Dapr will "see" expired messages.

Reply:
So if I publish a message with ttlInSeconds = 1, will I be guaranteeing that it will never be delivered?

Reply:
I'm unclear with exactly what is meant by
```
Dapr will add 1 second to allow component implementations to expire the message before Dapr consumes it.
```
Do I have this backwards? Will we end up delivering messages that should have been expired?

Reply:
If we were to use ms precision, would we be able to avoid this fudging of the timestamp?

Reply:
This has nothing to do with clock precision. We use `time.Now()`, which as per Golang's documentation: `A Time represents an instant in time with nanosecond precision.`

The 1s delay to expire is to avoid a situation where the component and Dapr are in a "race" to expire the message. Imagine that there is a ttl of 1 minute and th at message is being published to ServiceBus with a ttl of 1 minute. Dapr with calculate an expiration of now() + 1 minute and pass that message to the ServiceBus component that will publish the event with a ttl param value of 1 minute. Now, will Dapr get the message right before ServiceBus expires it or will Service Bus expire the message first? The 1s delay is to give Dapr a second priority when both the component and Dapr  can expire a message.

Reply:
This logic is gone now. If component implements message TTL feature, Dapr does not.